gnocchi/carbonara.py

```diff
@@ -453,11 +453,8 @@ def __iter__(self):
         return self
 
     def __hash__(self):
-        return hash(
-            str(datetime64_to_epoch(self.key))
-            +
-            str(self.sampling / ONE_SECOND)
-        )
+        return hash(str(self.key.astype('datetime64[ns]')) +
```

Comment:
i don't think this works as expected. from what i read, it's just tries to reevaluate based on bytes. it is faster though.

```
In [62]: dt.view()
Out[62]: numpy.datetime64('2017-01-01')

In [63]: dt.view('datetime64[ns]')
Out[63]: numpy.datetime64('1970-01-01T00:00:00.000017167')

In [64]: dt.astype('datetime64[ns]')
Out[64]: numpy.datetime64('2017-01-01T00:00:00.000000000')
```